kolibri/plugins/coach/assets/src/views/common/notifications/__test__/NotificationCard.spec.js

```diff
@@ -1,6 +1,8 @@
 import { mount, RouterLinkStub } from '@vue/test-utils';
 import NotificationCard from '../NotificationCard';
 
+NotificationCard.methods.getRoute = x => x;
```

Comment:
vue-test-utils v1 doesn't support overwriting methods when mounted. The same is true for `setMethods`. To solve this issue I used [this strategy](https://github.com/learningequality/kolibri/blob/13445ab3e443d381154b5cde44b644cf2bc5a0d1/kolibri/plugins/user/assets/test/views/profile-page.spec.js#L9-L11) for now though I'd like to open a follow-up to try to get rid of overwriting methods in this way in the whole codebase. It should be noted that it might be complicated at some places (see [this conversation](https://learningequality.slack.com/archives/CB37UM23A/p1594049471466000) to get more details) though maybe we can try to at least get rid of as much occurrences as possible.

Reply:
Yeah, I think most of the time I've done this is out of desperation and deactivate a method that has some kind AJAX side effect. So for most of these we might be able to use the xhr-mock library that was added recently. But for this particular case, I think it's okay to do it occasionally with the intention of mocking it properly later on so we actually cover the actual lines of code.